### PR TITLE
{lang}[foss/2021a] R v4.1.0

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

Requires:
* [x] ICU (~#13022~)
* [x] Tk (~#13023~)
* [x] GSL (~#13024~)
* [x] nodejs (~#13025~)
* [x] UDUNITS (~#13026~)
* [x] Ghostscript, ImageMagick, LittleCMS (~#13027~)
* [x] GDAL and deps (~#13030~)
* [x] Xvfb (~#13031~)

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/5c4d3adfa8437e0dad269b58dd46fdc0 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/0be77ecd737f4e99facb84e6dba92ca6 for a full test report.

**Comment by branfosj** [comment]
```
Error: package or namespace load failed for mclust in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs/mclust.so':
  /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs/mclust.so: undefined symbol: _ZGVbN2v_log
```

Also seen in https://github.com/easybuilders/easybuild-easyconfigs/pull/11455#issuecomment-707183596

**Comment by branfosj** [comment]
Setting
```
preconfigopts = 'export FLIBS="$FLIBS -lm" && '
```
in the ec solves the issue - that is putting that on the main R install. I've not found an equivalent for the individual extension to solve the issue.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/eb3f9a1dce8da0876637e59def55f90e for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/0367320c6b08701dc8d8c1dc7c033b05 for a full test report.

**Comment by boegel** [comment]
> Setting
> 
> ```
> preconfigopts = 'export FLIBS="$FLIBS -lm" && '
> ```
> 
> in the ec solves the issue - that is putting that on the main R install. I've not found an equivalent for the individual extension to solve the issue.

What about a patch file for `mclust`, similar to the ones we have for other extensions, see for example https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/r/R/optmatch-0.9-10_icpc-wd308.patch ?

File: easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,2921 @@
...
+    ('UDUNITS', '2.2.28'),  # for units
+    ('GSL', '2.7'),  # for RcppGSL
+    ('ImageMagick', '7.0.11-14'),  # for animation
+    ('GLPK', '5.0'),  # for Rglpk
+    ('nodejs', '14.17.0'),  # for V8 (required by rstan)
+    ('GDAL', '3.3.0'),  # for sf
+    ('OpenSSL', '1.1', '', True),
+]
+
+preconfigopts = 'export FLIBS="$FLIBS -lm" && '
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:56]: At the very least a comment should be added above to explain why this is needed, but it should be possible to come up with something specific to `mclust`...

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb]: I've removed this line and gone with the patch option, as per https://github.com/easybuilders/easybuild-easyconfigs/pull/13032#issuecomment-855378649 - done in cd4cd5e and c0a7dea

Comment (branfosj) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:58]: Scratch that. We hit a similar error on `quantreg` and `waveslim`. I'd expect it for other extensions as well. I've reverted and added a comment in 88c0e22.

Comment (boegel) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:58]: @branfosj I tried installing this easyconfig without the top-level `preconfigopts`, and it works fine for me...

Can you share the error output you're getting for one of these extensions, along with the failing command, so I can compare?

On which OS are you seeing trouble?

Comment (branfosj) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:58]: RHEL 8.3

Error is: `/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs/mclust.so: undefined symbol: _ZGVbN2v_log`

Build for `mclust` is:
```
== 2021-06-03 19:08:22,248 filetools.py:1775 INFO Creating directory /dev/shm/build-branfosj-admin/branfosj-admin-up/R/4.1.0/foss-2021a/mclust (parents: True, set_gid: False, sticky: False)
== 2021-06-03 19:08:22,248 run.py:226 INFO running cmd: tar xzf /rds/bear-sysadmin/configmgmt/easybuild/sources/r/R/extensions/mclust_5.4.7.tar.gz 
== 2021-06-03 19:08:22,349 run.py:226 INFO running cmd:  R CMD INSTALL /dev/shm/build-branfosj-admin/branfosj-admin-up/R/4.1.0/foss-2021a/mclust/mclust   --library=/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library --no-clean-on-error  
== 2021-06-03 19:08:45,924 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): cmd " R CMD INSTALL /dev/shm/build-branfosj-admin/branfosj-admin-up/R/4.1.0/foss-2021a/mclust/mclust   --library=/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library --no-clean-on-error " exited with exit code 1 and output:
* installing *source* package mclust ...
** package mclust successfully unpacked and MD5 sums checked
** using staged installation
** libs
gfortran -fno-optimize-sibling-calls  -fpic  -O2 -ftree-vectorize -march=native -fno-math-errno  -c dmvnorm.f -o dmvnorm.o
gcc -I"/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/include" -DNDEBUG   -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenSSL/1.1/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GDAL/3.3.0-foss-2021a/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/nodejs/14.17.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GLPK/5.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ImageMagick/7.0.11-14-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GSL/2.7-GCC-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/UDUNITS/2.2.28-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/HDF5/1.10.7-gompi-2021a/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ICU/69.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libsndfile/1.0.31-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FFTW/3.3.9-gompi-2021a/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/NLopt/2.7.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GMP/6.2.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libxml2/2.9.10-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cURL/7.76.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Tk/8.6.11-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Java/11.0.2/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/LibTIFF/4.2.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libjpeg-turbo/2.0.6-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libpng/1.6.37-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/PCRE2/10.36-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/SQLite/3.35.4-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/zlib/1.2.11-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/XZ/5.2.5-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/bzip2/1.0.8-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ncurses/6.2-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libreadline/8.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cairo/1.16.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libGLU/9.0.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Mesa/21.1.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/X11/20210518-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FlexiBLAS/3.0.4-GCC-10.3.0/include   -fpic  -O2 -ftree-vectorize -march=native -fno-math-errno  -c init.c -o init.o
gfortran -fno-optimize-sibling-calls  -fpic  -O2 -ftree-vectorize -march=native -fno-math-errno  -c mclust.f -o mclust.o
gfortran -fno-optimize-sibling-calls  -fpic  -O2 -ftree-vectorize -march=native -fno-math-errno  -c mclustaddson.f -o mclustaddson.o
gcc -shared -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenSSL/1.1/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenSSL/1.1/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GDAL/3.3.0-foss-2021a/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GDAL/3.3.0-foss-2021a/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/nodejs/14.17.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/nodejs/14.17.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GLPK/5.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GLPK/5.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ImageMagick/7.0.11-14-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ImageMagick/7.0.11-14-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GSL/2.7-GCC-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GSL/2.7-GCC-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/UDUNITS/2.2.28-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/UDUNITS/2.2.28-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/HDF5/1.10.7-gompi-2021a/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/HDF5/1.10.7-gompi-2021a/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ICU/69.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ICU/69.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libsndfile/1.0.31-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libsndfile/1.0.31-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FFTW/3.3.9-gompi-2021a/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FFTW/3.3.9-gompi-2021a/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/NLopt/2.7.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/NLopt/2.7.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GMP/6.2.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GMP/6.2.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libxml2/2.9.10-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libxml2/2.9.10-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cURL/7.76.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cURL/7.76.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Tk/8.6.11-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Tk/8.6.11-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Java/11.0.2/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Java/11.0.2/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/LibTIFF/4.2.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/LibTIFF/4.2.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libjpeg-turbo/2.0.6-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libjpeg-turbo/2.0.6-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libpng/1.6.37-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libpng/1.6.37-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/PCRE2/10.36-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/PCRE2/10.36-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/SQLite/3.35.4-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/SQLite/3.35.4-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/zlib/1.2.11-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/zlib/1.2.11-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/XZ/5.2.5-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/XZ/5.2.5-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/bzip2/1.0.8-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/bzip2/1.0.8-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ncurses/6.2-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ncurses/6.2-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libreadline/8.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libreadline/8.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cairo/1.16.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cairo/1.16.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libGLU/9.0.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libGLU/9.0.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Mesa/21.1.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Mesa/21.1.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/X11/20210518-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/X11/20210518-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Xvfb/1.20.11-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Xvfb/1.20.11-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ScaLAPACK/2.1.0-gompi-2021a-fb/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ScaLAPACK/2.1.0-gompi-2021a-fb/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FlexiBLAS/3.0.4-GCC-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FlexiBLAS/3.0.4-GCC-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GCCcore/10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GCCcore/10.3.0/lib -o mclust.so dmvnorm.o init.o mclust.o mclustaddson.o -lflexiblas -lgfortran -lgfortran -lgfortran -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/lib -lR
installing to /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for mclust in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs/mclust.so':
  /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs/mclust.so: undefined symbol: _ZGVbN2v_log
Error: loading failed
Execution halted
ERROR: loading failed
 (at easybuild/tools/run.py:565 in parse_cmd_output)
```

Comment (boegel) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:58]: Linking command looks identical on CentOS 7, no problem there. I'll also check on RHEL8

Comment (branfosj) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:58]: I'm also seeing this on our Ubuntu 20.04 (singularity container install).

There are other options for placing the `-lm` - I can add it to `LDFLAGS` instead and it still works. However, it does look like it needs setting in the R ec for me, as these items get stored in a config file inside the R install.

Comment (boegel) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:58]: I can confirm I see the same problem on RHEL8, the `-lm` needs to be specified explicitly.

I'm not sure whether it makes more sense to add it through `LDFLAGS`...

Comment (boegel) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:58]: The contents of `<installdir>/lib/R/etc/Makeconf` looks identical on both CentOS 7 and RHEL8 (except for absolute paths).

```
FLIBS = -lgfortran
```

Is there in both, yet somehow `-lm` is added automagically, or not needed at all, when linking `mclust` on CentOS 7?!

Comment (branfosj) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:58]: CentOS 7 brings in further differences. The missing symbols (`_ZGVbN2v_log`) are part of the vectorised math library (libmvec) so something different could be happening there. It confuses me that I've not seen this before.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 38 out of 41 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/d5a35e82aeae16457fc913d03f49335e for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/e02ec9d5c6a74e0f6c1f78a2944a960c for a full test report.

**Comment by boegel** [comment]
```
Checksum verification for extension source rda_1.0.2-2.1.tar.gz failed
```

Is that broken download, or are they really doing in-place updates again in CRAN?

**Comment by branfosj** [comment]
> 
> 
> ```
> Checksum verification for extension source rda_1.0.2-2.1.tar.gz failed
> ```
> 
> @verdurin Is that broken download, or are they really doing in-place updates again in CRAN?

:(
Sorry - I missed that the old easyconfigs had an alternative checksum for rda. I've readded that in 06382ce.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.4
See https://gist.github.com/606731cbb2383e3f671f979f1e38674e for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/bf4a5372a5388be92b6ab177763d7e32 for a full test report.

**Comment by verdurin** [comment]
@branfosj @boegel could you re-test on RHEL8 systems?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/4eab9592426a3763ee8c1b53721869c8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node2617.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/0887e1332f660a82389d258a1f0f5092 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/2e012e1b4f627c2652bb99640f79d0fa for a full test report.

**Comment by boegel** [comment]
@branfosj Looks like the `gert` extension need `git` 2.x?

```
----------------------------- ANTICONF -------------------------------
Configuration failed to find libgit2 library. Try installing:
 * brew: libgit2 (MacOS)
 * deb: libgit2-dev (Debian, Ubuntu, etc)
 * rpm: libgit2-devel (Fedora, CentOS, RHEL)
If libgit2 is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a libgit2.pc file. If pkg-config
is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
-------------------------- [ERROR MESSAGE] ---------------------------
<stdin>:1:10: fatal error: git2.h: No such file or directory
compilation terminated.
```

see https://github.com/easybuilders/easybuild-easyconfigs/pull/13073

**Comment by branfosj** [comment]
Added in 35458d4.

We see the failure because
```
Error in curl::curl_download("https://r-lib.github.io/gert/get-libgit2-linux.sh",  : 
          HTTP error 503.
```
which was when lots of the internet was down.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/de19b2d3e7e0497df5f518649fd97cef for a full test report.

File: easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb [hunk 2]
```diff
@@ -0,0 +1,2926 @@
...
+    ('libsndfile', '1.0.31'),  # for seewave
+    ('ICU', '69.1'),  # for rJava & gdsfmt
+    ('HDF5', '1.10.7'),  # for hdf5r
+    ('UDUNITS', '2.2.28'),  # for units
+    ('GSL', '2.7'),  # for RcppGSL
+    ('ImageMagick', '7.0.11-14'),  # for animation
+    ('GLPK', '5.0'),  # for Rglpk
+    ('nodejs', '14.17.0'),  # for V8 (required by rstan)
+    ('GDAL', '3.3.0'),  # for sf
+    ('git', '2.32.0', '-nodocs'),
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb:53]: @branfosj This should be `libgit2` rather than `git`...

```python
    ('libgit2', '1.1.0'),

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb (hunk 2 above)
Comment (branfosj) [reply easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb]: 1720ad2

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/5333d9113e8020aa5122cd6a614d9e8b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2617.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/512bfa21c8d9d2eead05e079ae959adb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/6a1e9a70a847cad2548452da3c10dc2f for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @branfosj!

**Pull request merged**